tokenizers/src/models/unigram/trainer.rs

```diff
@@ -154,6 +142,26 @@ impl UnigramTrainer {
         } else {
             (None, false)
         };
+
+        for (token, score) in model.iter() {
+            if inserted.contains::<str>(token) {
+                continue;
+            }
+            inserted.insert(token.to_string());
+            pieces.push((token.to_string(), if score.is_nan() { 0.0 } else { *score }));
+
+            let vocab_size_without_special_tokens = if need_add_unk {
+                self.vocab_size as usize - self.special_tokens.len() - 1
+            } else {
+                self.vocab_size as usize - self.special_tokens.len()
+            };
```

Comment:
Sorry I should have seen that earlier, do you mind extracting this from the loop (it can be defined on top.
This will likely be optimized away, but it still help readability too.

I can take care of it if you want.

Reply:
I should have found it out...